### DISCUSSION TITLE
Windows not recommended

### DISCUSSION THREAD

**Discussion by tooruu** [Ultimate-Hosts-Blacklist/Ultimate.Hosts.Blacklist#621] [Q&A]
Why is DNS preferred over hosts file on Windows?

**Comment by dnmTX** [comment]
@tooruu see #537 for specifics.

**Comment by spirillen** [comment]
The simple as in simple answer is, the hosts file is not cached in memory while DNS caching usually is =

1. RAM is faster than disk I/O
2. the hosts file remains a dumb flat file, while a DNS cache are using diff technique to be optimized
3. DNS cache uses the caching for optimizing the search for a reply
4. Hosts files a read from line 0 (1) to a line with a match if present AND then going to the DNS caching (resolver)

The simple zoomerize is, it is faster and it is better for your disk and network performance.

**Comment by zcwitkowski** [comment]
Actually you can use a HOSTS file in Windows, just maybe not as directly.

In Windows 98 SE a large HOSTS file seems to work OK. I had read that in later versions of Windows large HOSTS files are not a good idea, so, for XP I installed Acrylic DNS Proxy and it worked fine. The HOSTS file I used was mostly hpHosts with some other lists merged in ending up with 791404 entries the last time I was able to update it. If you want a local HOSTS file because you want to merge in additional lists you can tell Acrylic to use Ultimate Hosts Blacklist DNS *and* give it a local HOSTS file of your creation (or even multiple files). I don't know how well it works in later versions of Windows - one time I did try to buy a machine with Windows 10, but, it was just too horrible, I had to return it.

I have an Android 6 TV Box and an Android 11 tablet and I've been trying out DNS66 (got it from F-Droid). You can do DNS or blocklists or both. It's a lot easier to set up than Acrylic. Supposedly you can run it in an Android emulator on Windows, although, I haven't tried that myself.

**Comment by Ursus007de** [comment]
Hi, unfortunately I read the warning "not recommended for Windows" after I installed it and now I'm in trouble with W11. WLAN is connected, but no Internet access. 

I tryed to make the change reverse, but the hosts-file ist locked now (in use by a DNS-service) and neither editable nor replaceable (not manually, not with the script). It was not locked before the change. Maybe tooo big and somthing messed up.

What to do now? I deactivated the DNS Client Service (complicated way, because it is not stopable in an easy way. But now the trouble multiplied. 

Thank you for a short termed answer with a solution to replace the big hosts with the original.

**Reply by Ursus007de** [reply]
Could fix it for my own: Safe mode in windows and with terminal I could replace the big hosts file with the original one.

**Reply by spirillen** [reply]
Unfortunately that is very very bad to blacklist via the hosts file, and as you experience, you are getting trouble with the network connection, and this is simply because the hosts file is being read as the first source, this means the entire file a read, not cached, for each and very simple network package that are handled, each packet are by default 1500k.

The best solution, is if you can setup a Linux in a individual box and install a firewall like IPsec or alternatively your own DNS recursor like PowerDNS recursor + dnsdist, this mix gives you through control over dns lookup.

Alternatively you can use unbound for windows on your localhost, it gives you somewhat support for RPZ, but it can read and cache the huge hosts file without slowing down your network, but actually increase the dns lookup speed as the dns response delay will be optimized to something like 0-2 msec per query vs 70-150+ msec as you have now.

If you need help installing any of this, please reach out, before you god your self locked out
